src/App.jsx

```diff
@@ -16,13 +25,25 @@ function App() {
     setNum(randomNumberInRange(1, 6));
     setNum2(randomNumberInRange(1, 6));
     setScore(score+num+num2);
+    if(score>=100) {
+      disableButton();
+    }
+    else {
+      enableButton();
+    };
+    if(score==100){
+      alert("Congratulations, you won!");
+    };
+    if(score>100){
+      alert("You lose!");
+    };
   };
   return (
     <div>
       <div>dice 1: {num}</div>
       <div>dice 2: {num2}</div>
       <div>score= {score}</div>
-      <button onClick={handleClick}>Roll dice</button>
+      <button onClick={handleClick} disabled={isButtonDisabled}>Roll dice</button>
```

Comment:
```suggestion
      <button onClick={handleClick} disabled={score >= 100}>Roll dice</button>
```
Du könntest es sogar noch kürzer ohne den extra state machen :)